atlite/cutout.py

```diff
@@ -1,120 +1,177 @@
+# resampling the sarah dataset it is important to use conda.
+# Refer to
+# https://github.com/pydata/xarray/issues/2535,
+# https://github.com/rasterio/rasterio-wheels/issues/12
+import netCDF4
 
 import xarray as xr
 import numpy as np
 import os, sys
-from six import string_types
+from warnings import warn
+from shapely.geometry import box
+from pathlib import Path
 
 import logging
 logger = logging.getLogger(__name__)
 
-from . import config, datasets
+from . import datasets, utils
 
 from .convert import (convert_and_aggregate, heat_demand, hydro, temperature,
                       wind, pv, runoff, solar_thermal, soil_temperature)
-from .preparation import (cutout_do_task, cutout_prepare,
-                          cutout_produce_specific_dataseries,
-                          cutout_get_meta, cutout_get_meta_view)
-from .gis import compute_indicatormatrix
-
-class Cutout(object):
-    def __init__(self, name=None, cutout_dir=config.cutout_dir, **cutoutparams):
-        self.name = name
+from .gis import GridCells
+from .data import requires_coords, requires_windowed, cutout_prepare
+
+class Cutout:
+
+    dataset_module = None
+
+    def __init__(self, path=None, data=None, cutout_dir=None, **cutoutparams):
+        """
+        Provide an Atlite cutout object.
+
+        Create a cutout object to use atlite operations on it.
+        Based on the provided parameters, atlite first checks
+        whether this cutout already exists on disk and if yes,
+        loads this cutout.
+
+        If the cutout does not yet exist on disk, then atlite
+        creates an "unprepared" cutout object containing all the
+        necessary information for creating the requested cutout,
+        but does not yet create ("prepare") it fully.
+        The process of preparing has to be manually started by
+        `cutout.prepare()`.
+
+        Parameters
+        ----------
+        path : str | path-like
+            NetCDF from which to load or where to store the cutout
+        data : (opt.) xr.Dataset
+            Specify the NetCDF data directly
+        module : ["era5","ncep","cordex","sarah"]
+            The dataset which works as a basis for the cutout
+            creation.
+        time : str | slice
+            Time range to include in the cutout, e.g. "2011" or
+            ("2011-01-05", "2011-01-25")
+        bounds : (opt.) GeoSeries.bounds | DataFrame
+            The outer bounds of the cutout or as a DataFrame
+            containing (min.long, min.lat, max.long, max.lat).
+        x : (opt.) slice
+            Outer longitudinal bounds for the cutout (west, east).
+        y : (opt.) slice
+            Outer latitudinal bounds for the cutout (south, north).
+        time : (opt.) slice
+
+        """
+        if cutout_dir is not None:
+            warn("The argument `cutout_dir` has been deprecated in favour of just `path`", DeprecationWarning)
+            path = Path(cutout_dir) / path
```

Comment:
```suggestion
            path = Path(cutout_dir) 
```
This raised an error with my old code

Reply:
Could you provide the error raised + `atlite.Cutout(...)` call signature you used?

Reply:
Yes of course, I think I understand why you're hesitating. In my case `cutout_dir`  was a string and path was None (default). Then it tries to append `None` to the `cutout_dir` which is not working:
```python
  File "/home/fabian/vres/py/atlite/atlite/cutout.py", line 82, in __init__
    path = Path(cutout_dir) / path

  File "/home/fabian/.miniconda3/lib/python3.7/pathlib.py", line 920, in __truediv__
    return self._make_child((key,))

  File "/home/fabian/.miniconda3/lib/python3.7/pathlib.py", line 699, in _make_child
    drv, root, parts = self._parse_args(args)

  File "/home/fabian/.miniconda3/lib/python3.7/pathlib.py", line 653, in _parse_args
    a = os.fspath(a)

TypeError: expected str, bytes or os.PathLike object, not NoneType
```
The intention of this line is probably to select a nc file within the `cutout_dir`?

Reply:
Probably to keep it backwards compatible to when we were using `cutout_dir´ and `name` rather than `path`.

I.e. this should be `name`, not `path`.

Reply:
This should  now be fixed.